In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from pylab import rcParams

rcParams["figure.figsize"] = (10, 6)
sns.set()

In [2]:
main_data = pd.read_csv("FuelConsumptionCo2.csv")

In [3]:
main_data.sample(5)

MODELYEAR       MAKE                  MODEL             VEHICLECLASS  \
81        2014        BMW                   328i                  COMPACT   
485       2014      HONDA                 ACCORD                 MID-SIZE   
902       2014    PORSCHE             PANAMERA 4                FULL-SIZE   
243       2014  CHEVROLET          SILVERADO 4WD  PICKUP TRUCK - STANDARD   
985       2014     TOYOTA  HIGHLANDER HYBRID AWD           SUV - STANDARD   

     ENGINESIZE  CYLINDERS TRANSMISSION FUELTYPE  FUELCONSUMPTION_CITY  \
81          2.0          4           A8        Z                  10.0   
485         3.5          6          AS6        X                  11.4   
902         3.6          6          AM7        Z                  13.1   
243         4.3          6           A6        E                  20.0   
985         3.5          6          AV6        X                   8.0   

     FUELCONSUMPTION_HWY  FUELCONSUMPTION_COMB  FUELCONSUMPTION_COMB_MPG  \
81                   6.5                   8.4                        34   
485                  7.3                   9.6                        29   
902                  8.7                  11.1                        25   
243                 15.1                  17.8                        16   
985                  8.6                   8.3                        34   

     CO2EMISSIONS  
81            193  
485           221  
902           255  
243           285  
985           191

### Step 1

In [4]:
main_data[["FUELCONSUMPTION_CITY", "FUELCONSUMPTION_HWY", "FUELCONSUMPTION_COMB", "FUELCONSUMPTION_COMB_MPG", "CO2EMISSIONS"]].describe()

FUELCONSUMPTION_CITY  FUELCONSUMPTION_HWY  FUELCONSUMPTION_COMB  \
count           1067.000000          1067.000000           1067.000000   
mean              13.296532             9.474602             11.580881   
std                4.101253             2.794510              3.485595   
min                4.600000             4.900000              4.700000   
25%               10.250000             7.500000              9.000000   
50%               12.600000             8.800000             10.900000   
75%               15.550000            10.850000             13.350000   
max               30.200000            20.500000             25.800000   

       FUELCONSUMPTION_COMB_MPG  CO2EMISSIONS  
count               1067.000000   1067.000000  
mean                  26.441425    256.228679  
std                    7.468702     63.372304  
min                   11.000000    108.000000  
25%                   21.000000    207.000000  
50%                   26.000000    251.000000  
75%                   31.000000    294.000000  
max                   60.000000    488.000000

##### So here we have some quantitive overview of some of variables

### Step 2

##### Observation:
- Average Co2 Emissions is 256.23 and minimum is 108 and maximum emissions is 488
- Here Fuel Consumption is directly affect on Co2 Emissions
- So if there is less Fuel Consumption then co2 Emissions will be less.